ocs_ci/ocs/container.py

```diff
@@ -0,0 +1,315 @@
+
+import logging
+import time
+import concurrent.futures
+
+from ocs_ci.ocs import constants, defaults
+from ocs_ci.utility import templating
+from ocs_ci.ocs.resources.ocs import OCS
+from ocs_ci.utility.utils import run_cmd
+from ocs_ci.ocs.exceptions import CommandFailed
+
+from tests.helpers import (create_unique_resource_name,
+                           get_admin_key, get_cephfs_name,
+                           get_cephfs_data_pool_name)
+
+log = logging.getLogger(__name__)
+
+
+class Container(object):
+    """
+    Container Storage class
+
+    Args:
+        interface_type (str): type of interface to use RBD or FS
+        block_pool_name (str): name of the ceph block pool name
+        reclaim_policy (str): pvc recliam policy (retain/delete)
+        sc_name (str): Storage class metadata name
+        pvc_name (str): PVC metadata name
+        num_pvc (int): Number of PVCs to create
+        access_mode (str): Access mode for PVC - RWO/RWX/ROX
+        namespace (str): namespace for the PVC
+        wait (bool): whether to wait for PVC to be in BOUND state
+        size (int): Size of the PVC in GB
+        cleanup (bool): If true, the previously created objects are cleanedup
+
+    """
+    def __init__(self,
+                 interface_type=constants.CEPHBLOCKPOOL,
+                 block_pool_name=None,
+                 reclaim_policy=constants.RECLAIM_POLICY_DELETE,
+                 sc_name=None,
+                 pvc_name=None,
+                 num_pvc=1,
+                 access_mode=constants.ACCESS_MODE_RWO,
+                 namespace=defaults.ROOK_CLUSTER_NAMESPACE,
+                 wait=True,
+                 size=None,
+                 cleanup=False,
+                 ):
+        self.interface_type = interface_type
+        self.pool_name = block_pool_name
+        self.sc_name = sc_name
+        self.pvc_name = pvc_name
+        self.num_pvc = num_pvc
+        self.access_mode = access_mode
+        self.reclaim_policy = reclaim_policy
+        self.namespace = namespace
+        self.size = size
+        self.wait = wait
+        self.sc_data = dict()
+        self.block_data = dict()
+        self.secret_data = dict()
+        self.pvc_data = dict()
+        if cleanup:
+            self._cleanup()
+        if interface_type == constants.CEPHBLOCKPOOL:
+            self._setup_rbd_secret()
+            self._setup_ceph_blockpool()
+            self._setup_rbd_storageclass()
+        else:
+            self._setup_fs_secret()
+            self._setup_fs_storageclass()
```

Comment:
storage class raw block pvc should be considered too (maybe as a later patch). This is applicable wherever if..else is used below to create rbd or cephfs resources.

Reply:
I am not sure I got you question, but both raw block/fs exists in this class.